### PR TITLE
fix: text alignment should be based on numeric or not numeric

### PR DESCRIPTION
**Pull request by LiKang6688**
- head cell 
	- column is measure: right
	- column is dimension
		- 	qDimensionType is N: right
		- otherwise: left

- totals cell
	- column is dimension: left
	- column is measure right

- inner body cell
	- cell's qNum is number: right
	- cell's qNum is not number: left

user-set text alignment should override the above.

### CODE REVIEW

File: src/handle-data.ts [hunk 1]
```diff
@@ -79,7 +81,9 @@ export function getColumnInfo(layout: TableLayout, colIdx: number, pageColIdx: n
   } = info;
   const isHidden = qError?.qErrorCode === 7005;
   const isLocked = isDim && (info as ExtendedNxDimensionInfo).qLocked;
-  const autoAlign = isDim ? 'left' : 'right';
+  const { qDimensionType } = info as ExtendedNxDimensionInfo;
+  const autoHeadCellTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
+  const autoTotalsCellTextAlgin = isDim ? 'left' : 'right';
```
Comment (haxxmaxx) [inline src/handle-data.ts:86]: typo

File: src/handle-data.ts [hunk 2]
```diff
@@ -101,11 +105,16 @@ export function getColumnInfo(layout: TableLayout, colIdx: number, pageColIdx: n
...
       stylingIDs: qAttrExprInfo.map((expr) => expr.id),
       // making sure that qSortIndicator is either A or D
       sortDirection: qSortIndicator && qSortIndicator !== 'N' ? qSortIndicator : 'A',
       totalInfo: getTotalInfo(layout, colIdx, pageColIdx, numDims),
+      autoHeadCellTextAlign,
+      autoTotalsCellTextAlgin,
+      textAlign: {
+        auto: !textAlign || textAlign.auto,
+        align: textAlign?.align,
+      },
```
Comment (haxxmaxx) [inline src/handle-data.ts:117]: I don't really see why we need all of this information for later. Wouldn't it be possible to figure out what the final align is for all three cases (head, totals, body) and store those three straight on the column

File: src/table/virtualized-table/Cell.tsx [hunk 3]
```diff
@@ -18,21 +19,19 @@ interface CellProps {
 
 const Cell = ({ columnIndex, rowIndex, style, data }: CellProps) => {
   const { rowsInPage, columns, bodyStyle, isHoverEnabled, maxLineCount } = data;
-  const cell = rowsInPage[rowIndex]?.[`col-${columnIndex}`];
+  const cell = rowsInPage[rowIndex]?.[`col-${columnIndex}`] as CellType;
```
Comment (haxxmaxx) [inline src/table/virtualized-table/Cell.tsx:22]: not sure you can cast this, @cbt1 can it be undefined at this stage?

File: src/table/pagination-table/components/body/TableBodyWrapper.tsx [hunk 4]
```diff
@@ -61,8 +61,10 @@ function TableBodyWrapper({
           className="sn-table-row"
         >
           {columns.map((column, columnIndex) => {
-            const { id, align } = column;
+            const { id, textAlign } = column;
             const cell = row[id] as Cell;
+            const autoAlign = () => (cell.qNum && !Number.isNaN(+cell.qNum) ? 'right' : 'left');
+            const align = textAlign.auto ? autoAlign() : textAlign.align;
```
Comment (haxxmaxx) [inline src/table/pagination-table/components/body/TableBodyWrapper.tsx:67]: This we also could do on a handle-data level, when we create `rows`. Unfortunately we have do do it in a different place for the virtual scroll. But we could reuse the logic.

Comment (LiKang6688) [reply src/table/pagination-table/components/body/TableBodyWrapper.tsx:67]: Yes, that is how the old table behaves
Screenshot 2023-03-09 at 09 48 34

File: src/handle-data.ts (hunk 2 above)
Comment (LiKang6688) [reply src/handle-data.ts:117]: we need data cell level. I do not see the easy solution to store them on the column

**Comment by LiKang6688** [comment]
gonna move logic from component level to fetch data place

**Comment by LiKang6688** [comment]
gonna move logic from component level to fetch data place

**Pull request closed without merging**